components/service/glean/src/main/java/mozilla/components/service/glean/histogram/FunctionalHistogram.kt

```diff
@@ -0,0 +1,185 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.histogram
+
+import mozilla.components.support.ktx.android.org.json.tryGetLong
+import org.json.JSONObject
+import java.lang.Math.pow
+import kotlin.math.log
+
+/**
+ * This class represents a histogram where the bucketing is performed by a
+ * function, rather than pre-computed buckets. It is meant to help serialize
+ * and deserialize data to the correct format for transport and storage, as well
+ * as performing the calculations to determine the correct bucket for each sample.
+ *
+ * The bucket index of a given sample is determined with the following function:
+ *
+ *     i = ⌊n log₂(𝑥)⌋
+ *
+ * In other words, there are n buckets for each power of 2 magnitude.
+ *
+ * The value of 8 for n was determined experimentally based on existing data to have sufficient
```

Comment:
Shouldn't the coefficient be 8 for timing and 16 for memory? See [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1565253#c5)

Reply:
Oh, thanks for catching that -- I hadn't seen Corey's response...

This will require some retooling to make parameterize that...

Reply:
nit: "The value of 8" needs update, since that is now parametrized